### PR TITLE
Pass same multiprocessing settings to evaluate as to fit

### PR DESCRIPTION
**Pull request by aamster**
The call to `model.evaluate` when evaluating initial validation loss when finetuning uses the default settings for multiprocessing. This means multiprocessing is disabled.

This has significant implications for the time it takes to do a forward pass of the validation set. 

In a small test, it took 37 seconds to call `model.evaluate` with no multiprocessing (current setting) compared with 17 seconds with this PR.

This has big implications when the data size is big. To evaluate 40,000 examples, it took 48 hours in the initial `model.evaluate` call and only 9 hours in the `model.fit` call with multiprocessing. This validation size was too big, but still it shows the issue.

### CODE REVIEW

**Review by jeromelecoq — APPROVED** [review]
Good catch
